test-vfkit/basic_test.go

```diff
@@ -0,0 +1,223 @@
+package e2e_vfkit
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	gvproxyclient "github.com/containers/gvisor-tap-vsock/pkg/client"
+
+	"github.com/containers/gvisor-tap-vsock/pkg/types"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+var _ = ginkgo.Describe("connectivity with vfkit", func() {
```

Comment:
Came back to this now... I think your comment is not valid anymore after having agreed to using different dirs? Because if we have different BeforeSuite on each hypervisor directory I don't think we can use the same basic_test file right? or you have an idea how to achieve it?

Reply:
I agree with you it would be great to remove all that duplicate code if we find a way

Reply:
Found a way to share the basic tests. So basically I put the `It` clauses inside a func in a shared file and then they are called inside the `Describe` callback on the actual hypervisor folder where the tests has to run.

If you have a better idea how to do this, I'm all ears 👍